src/arch/wasm/CodeGen.zig

```diff
@@ -6446,63 +6446,89 @@ fn airDivFloor(func: *CodeGen, inst: Air.Inst.Index) InnerError!void {
     const rhs = try func.resolveInst(bin_op.rhs);
 
     if (ty.isUnsignedInt(mod)) {
-        const result = try (try func.binOp(lhs, rhs, ty, .div)).toLocal(func, ty);
-        return func.finishAir(inst, result, &.{ bin_op.lhs, bin_op.rhs });
+        _ = try func.binOp(lhs, rhs, ty, .div);
     } else if (ty.isSignedInt(mod)) {
         const int_bits = ty.intInfo(mod).bits;
         const wasm_bits = toWasmBits(int_bits) orelse {
-            return func.fail("TODO: `@divFloor` for signed integers larger than '{d}' bits", .{int_bits});
+            return func.fail("TODO: `@divFloor` for signed integers larger than 64 bits ({d} bits requested)", .{int_bits});
         };
-        const lhs_res = if (wasm_bits != int_bits) blk: {
-            break :blk try (try func.signAbsValue(lhs, ty)).toLocal(func, ty);
-        } else lhs;
-        const rhs_res = if (wasm_bits != int_bits) blk: {
-            break :blk try (try func.signAbsValue(rhs, ty)).toLocal(func, ty);
-        } else rhs;
+
+        if (wasm_bits > 64) {
+            return func.fail("TODO: `@divFloor` for signed integers larger than 64 bits ({d} bits requested)", .{int_bits});
+        }
+
+        const lhs_wasm = if (wasm_bits != int_bits)
+            try (try func.signExtendInt(lhs, ty)).toLocal(func, ty)
+        else
+            lhs;
+
+        const rhs_wasm = if (wasm_bits != int_bits)
+            try (try func.signExtendInt(rhs, ty)).toLocal(func, ty)
+        else
+            rhs;
 
         const zero = switch (wasm_bits) {
             32 => WValue{ .imm32 = 0 },
             64 => WValue{ .imm64 = 0 },
-            else => unreachable,
+            else => @panic("Unexpected wasm integer width"),
```

Comment:
No `@panic` in the backend, please. The `unreachable` notates it would be a developer error if reached. Line #6456 you added already provides a compilation error with a TODO. Once implemented, the implementer must update the branch also. If a comment is added to the line, after `unreachable,` it will also show up in the stack trace.

Reply:
Indeed. https://github.com/ziglang/zig/pull/16807#issuecomment-1685298962

Reply:
Changed.